RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -886,13 +886,315 @@ double AntiElectronIDMVA6::MVAValue(const reco::PFTau& thePFTau)
                   0.);
 }
 
+double AntiElectronIDMVA6::MVAValue(const pat::Tau& theTau, const pat::Electron& theEle)
+{
+  // === tau variables ===
+  float TauEtaAtEcalEntrance = theTau.etaAtEcalEntrance();
+  
+  float TauLeadChargedPFCandEtaAtEcalEntrance = theTau.etaAtEcalEntranceLeadChargedCand();
+  float TauLeadChargedPFCandPt = theTau.ptLeadChargedCand();
+
+  Float_t TauPt = theTau.pt();
+  //Float_t TauEmFraction = std::max(theTau.ecalEnergy()/(theTau.ecalEnergy()+theTau.hcalEnergy()), (Float_t)0.);
+  Float_t TauEmFraction = std::max(theTau.emFraction_MVA(), (Float_t)0.);
+  Float_t TauLeadPFChargedHadrHoP = 0.;
+  Float_t TauLeadPFChargedHadrEoP = 0.;
+  if ( theTau.leadChargedHadrCand()->p() > 0. ) {
+    TauLeadPFChargedHadrHoP = theTau.hcalEnergyLeadChargedHadrCand()/theTau.leadChargedHadrCand()->p();
+    TauLeadPFChargedHadrEoP = theTau.ecalEnergyLeadChargedHadrCand()/theTau.leadChargedHadrCand()->p();
+  }
+
+  std::vector<Float_t> GammasdEtaInSigCone;
+  std::vector<Float_t> GammasdPhiInSigCone;
+  std::vector<Float_t> GammasPtInSigCone;
+  std::vector<Float_t> GammasdEtaOutSigCone;
+  std::vector<Float_t> GammasdPhiOutSigCone;
+  std::vector<Float_t> GammasPtOutSigCone;
+  reco::Candidate::LorentzVector pfGammaSum(0,0,0,0);
+  reco::Candidate::LorentzVector pfChargedSum(0,0,0,0);
+  
+  const reco::CandidatePtrVector signalGammaCands = theTau.signalGammaCands();
+  for ( reco::CandidatePtrVector::const_iterator gamma = signalGammaCands.begin(); gamma != signalGammaCands.end(); ++gamma ){
+    float dR = deltaR((*gamma)->p4(), theTau.leadChargedHadrCand()->p4());
+    float signalrad = std::max(0.05, std::min(0.10, 3.0/std::max(1.0, theTau.pt())));
+
+    // pfGammas inside the tau signal cone
+    if (dR < signalrad) {
+      if ( theTau.leadChargedHadrCand().isNonnull() ) {
+        GammasdEtaInSigCone.push_back((*gamma)->eta() - theTau.leadChargedHadrCand()->eta());
+        GammasdPhiInSigCone.push_back((*gamma)->phi() - theTau.leadChargedHadrCand()->phi());
```

Comment:
Hi Carl,

I agree with you. But since these variables are used that way in the current training I will leave them as they are. I've added a note to Anne-Catherine our developer on the anti-electron MVA to check and in case change for the next training to come, to be consistent in training and application.  

Cheers,
Roger